tools/wpt/browser.py

```diff
@@ -581,17 +581,16 @@ def version(self, binary=None, webdriver_binary=None):
         return m.group(1)
 
 
-class ChromeAndroid(Browser):
-    """Chrome-specific interface for Android.
+class ChromeAndroidBase(Browser):
```

Comment:
I don't see the need for creating this base class. `AndroidWebview` can directly inherit `ChromeAndroid`; `AndroidWebview` can directly override any methods or properties that are different (including `product` and `requirements`).

This is fine, too, but we might want to mark this base class as abstract (e.g. `__metaclass__ = abc.ABCMeta`).

Reply:
It is just a matter of how we conceptualize this, but I'm worried that future developers of this file would just change ChromeAndroid without trying to check its impact on AndroidWebView class. Therefore, I prefer to keep the common logic into a base class. I'll make this abstract as you suggested.